File: backend/cp25ssi2/src/main/java/sit/chat2date/cp25ssi2/dto/ReportResponse.java
package sit.chat2date.cp25ssi2.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import sit.chat2date.cp25ssi2.enums.ReportStatus;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReportResponse {
    private String reportId;
    private String reporterId;
    private String targetUserId;
    private String reason;
    private ReportStatus status;
    private LocalDateTime createdAt;
}

File: frontend-mobile/chat2date/lib/models/report_response.dart
class ReportResponse {
  final int reportId;
  final String reporterId;
  final String targetUserId;
  final String reason;
  final String status;
  final DateTime createdAt;

  ReportResponse({
    required this.reportId,
    required this.reporterId,
    required this.targetUserId,
    required this.reason,
    required this.status,
    required this.createdAt,
  });

  factory ReportResponse.fromJson(Map<String, dynamic> json) {
    return ReportResponse(
      reportId: json['reportId'] as int,
      reporterId: json['reporterId'] as String,
      targetUserId: json['targetUserId'] as String,
      reason: json['reason'] as String,
      status: json['status'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
